modules/swagger-parser-v3/src/main/java/io/swagger/parser/v3/util/OpenAPIDeserializer.java

```diff
@@ -1785,20 +1785,18 @@ public Schema getSchema(ObjectNode node, String location, ParseResult result){
         ObjectNode itemsNode = getObject("items", node, false, location, result);
 
         if(allOfArray != null) {
-
+            ComposedSchema allOfList = new ComposedSchema();
             for(JsonNode n : allOfArray) {
                 if(n.isObject()) {
-                    ComposedSchema allOfList = new ComposedSchema();
                     schema = getSchema((ObjectNode) n,location,result);
                     allOfList.addAllOfItem(schema);
                     schema = allOfList;
                 }
             }
         } else if(anyOfArray != null) {
-
+            ComposedSchema anyOfList = new ComposedSchema();
             for(JsonNode n : anyOfArray) {
                 if(n.isObject()) {
-                    ComposedSchema anyOfList = new ComposedSchema();
                     schema = getSchema((ObjectNode) n,location,result);
                     anyOfList.addAnyOfItem(schema);
                     schema = anyOfList;
```

Comment:
`schema = anyOfList;` : this line should be outside the `for` structure

Reply:
OMG It was.